File: ai-service/app/routers/health.py
"""
app/routers/health.py
Health-check endpoint for the FastAPI AI service.
"""

from datetime import datetime, timezone
from fastapi import APIRouter

router = APIRouter()


@router.get("/", summary="Health check", tags=["Health"])
def health_check():
    """
    Returns a simple JSON confirming the AI service is running.
    No image processing or OCR is triggered here.
    """
    return {
        "status": "ok",
        "service": "Label Lens — FastAPI AI Service",
        "version": "1.0.0",
        "timestamp": datetime.now(timezone.utc).isoformat(),
    }

File: backend/src/routes/health.js
/**
 * routes/health.js
 * Health-check endpoint for the Express backend.
 */

const express = require("express");
const router = express.Router();

// GET /
// Returns a simple JSON confirming the backend is running.
router.get("/", (req, res) => {
  res.json({
    status: "ok",
    service: "Label Lens — Express Backend",
    version: "1.0.0",
    timestamp: new Date().toISOString(),
  });
});

module.exports = router;
